lib/drives/pages/drive_chat_page.dart

```diff
@@ -18,130 +18,101 @@ class DriveChatPage extends StatefulWidget {
 
 class _DriveChatPageState extends State<DriveChatPage> {
   Drive? _drive;
-  bool _fullyLoaded = false;
 
   @override
   void initState() {
     super.initState();
     setState(() {
       _drive = widget.drive;
     });
-    loadDrive();
   }
 
   @override
   Widget build(BuildContext context) {
     List<Widget> widgets = [];
-    if (_fullyLoaded) {
-      Set<Profile> riders = _drive!.approvedRides!.map((ride) => ride.rider!).toSet();
-      List<Ride> pendingRides = _drive!.pendingRides!.toList();
-      if (riders.isEmpty && pendingRides.isEmpty) {
-        return Scaffold(
-          appBar: AppBar(
-            title: Text(S.of(context).pageDriveChatTitle),
-          ),
-          body: Center(
-            child: Text(S.of(context).pageDriveChatEmptyMessage),
-          ),
-        );
-      } else {
-        if (riders.isNotEmpty) {
-          List<Widget> riderColumn = [
-            const SizedBox(height: 5.0),
-            Text(
-              S.of(context).pageDriveChatRiderHeadline,
-              style: Theme.of(context).textTheme.titleLarge,
-            ),
-            const SizedBox(height: 10.0),
-            _riderList(riders),
-          ];
-          widgets.addAll(riderColumn);
-        }
-        widgets.add(const SizedBox(height: 10.0));
-        if (pendingRides.isNotEmpty) {
-          List<Widget> pendingRidesColumn = [
-            const SizedBox(height: 5.0),
-            Text(S.of(context).pageDriveChatRequestsHeadline, style: Theme.of(context).textTheme.titleLarge),
-            const SizedBox(height: 10.0),
-          ];
-          pendingRidesColumn.addAll(_pendingRidesList(pendingRides));
-          widgets.addAll(pendingRidesColumn);
-        }
-      }
+    Set<Ride> approvedRides = _drive!.approvedRides!.toSet();
+    List<Ride> pendingRides = _drive!.pendingRides!.toList();
+    if (approvedRides.isEmpty && pendingRides.isEmpty) {
+      return Scaffold(
+        appBar: AppBar(
+          title: Text(S.of(context).pageDriveChatTitle),
+        ),
+        body: Center(
+          child: Text(S.of(context).pageDriveChatEmptyMessage),
+        ),
+      );
     } else {
-      widgets.add(const SizedBox(height: 10));
-      widgets.add(const Center(child: CircularProgressIndicator()));
+      if (approvedRides.isNotEmpty) {
+        List<Widget> riderColumn = approvedRides.map((ride) => _buildChatWidget(ride)).toList();
+        widgets.addAll(riderColumn);
+      }
+      widgets.add(const SizedBox(height: 10.0));
     }
     return Scaffold(
       appBar: AppBar(
         title: Text(S.of(context).pageDriveChatTitle),
       ),
-      body: RefreshIndicator(
```

Comment:
I think a RefreshIndicator would still be helpful. Or even better, the page should immediately pick up new messages too...

Reply:
didn't we talk about that yesterday and you sad that you would remove it? But yeah i can add it. We could also add that the messages are picked up but then we would need another stream, or streams for each ride.

Reply:
I think a Stream would probably be best? It's what users would expect from a chat functionality.

 I don't know what I said yesterday lol - sorry if I caused any confusion.

Reply:
ok i'll try and do that

Reply:
okay i did that